frontend/app/controllers/spree/checkout_controller.rb

```
@@ -93,6 +91,63 @@ def rescue_from_spree_gateway_error(exception)
       render_unprocessable_entity
     end
 
+    # Update order from params
+    #
+    # @return [Boolean]
+    #   the validility of the resulting order
+    #
+    # @api private
+    def update_from_params
+      @order.update_attributes(checkout_attributes.merge(payments_attributes))
+    end
+
+    # Effective payment attributes
+    #
+    # @return [ActionController::Parameters]
+    def payments_attributes
+      attributes = checkout_attributes[:payments_attributes].try(:first)
```

Oh what objects can `checkout_attributes[:payments_attributes]` return that would not respond to `#first`?

If somehow `checkout_attributes[:payments_attributes]` returns `'1'`, then wouldn't we want to explode instead of silently ignore it?